applications/tari_console_wallet/src/ui/components/network_tab.rs

```diff
@@ -14,10 +14,41 @@ impl NetworkTab {
     pub fn new() -> Self {
         Self {}
     }
+
+    pub fn draw_connected_peers_list<B>(&self, f: &mut Frame<B>, area: Rect, app_state: &AppState)
+    where B: Backend {
+        let block = Block::default().borders(Borders::ALL).title(Span::styled(
+            "Connected Peers",
+            Style::default().fg(Color::White).add_modifier(Modifier::BOLD),
+        ));
+        f.render_widget(block, area);
+
+        let list_areas = Layout::default()
+            .constraints([Constraint::Min(1)].as_ref())
+            .margin(1)
+            .split(area);
+
+        let peers = app_state.get_connected_peers();
+        let mut column0_items = Vec::new();
+        let mut column1_items = Vec::new();
+        let mut column2_items = Vec::new();
```

Comment:
nit: single allocation each

```suggestion
        let mut column0_items = Vec::with_capacity(peers.len());
        let mut column1_items = Vec::with_capacity(peers.len());
        let mut column2_items = Vec::with_capacity(peers.len());
```